### PR TITLE
Bump Gtest 1.11

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
See commits for descriptions

Tl;dr switch `boost::optional` to `std::optional` in GUI Mocks to make gtest 1.11 work, so I can compile with new(ish) versions of clang to fix a race

**To test:**
- Check all unit tests pass

There is no associated issue.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DavidFair** [comment]
Closing this PR as it requires us to upgrade our compiler, which we can do later on
